### ISSUE TITLE
Missing field star catalogs

### ISSUE THREAD

**Issue by dave-kaye** [XEphem/Catalogs#1]
There are two field stars catalogs missing from the catalog repository.  The readme file in the catalog directory lists the field stars for Xephem 3.7 as:

2. Field Stars:
    a) USNO SA2.0 image Catalog.------------------------>  /usno
    b) GSC-ACT Catalog. --------------------------------->  /gsc
    c) Hipparcos and Tycho-2 Catalogues. ---------->  hiptyc2.xe2
    d) Positions in Proper Motions Catalog. ----------->  ppm.xe2

This has been increased by the addition of the GSC 2.2 catalog ----> /GSC2201

The hiptyc2.xe2 and ppm.xe2 files are located in the Official directory.  The GSC2201 catalog has its own directory.  But the USNO SA2.0 and GSC-ACT catalogs are not found.

THe GSC-ACT is an earlier and smaller version of the GSC2201.  It contains 18,839,509 entries vs the 451,632,833 entries of the more complex version.  It also is only 145 MB versus the over 1 GB size of the larger version.  It can be accessed using the guide star window --> Hubble Guide Star Catalog, GSC section --> Local Cache Directory.  Setting the directory to the location of the GSC-ACT directory will access these stars.  

The USNO-SA2.0 is according to the USNO "a subset of USNO-A2.0 which is a lot easier to handle on a small computer because it contains only a tenth as many stars as the parent catalog (54,787,624 stars in all). The goal in creating this smaller catalog was to provide a spatially uniform distribution of stars in an intermediate range of magnitudes which would be useful as a 'reference grid' for astrometric analysis."  Its size is 588 MB.

One of the Downey Xephem issues is updating field stars.  Leaving out these databases could be a part of this effort.  The GSC-ACT catalog being an older GSC, still appears to have decent accuracy with the corrections from the ACT.  It was superseded by the more comprehensive GSC 2.2.  It can be argued that its smaller size may have benefits for some users and should be considered to be retained.  The USNO SA2.0 is a very unique field star representation.  It would have utility due to the uniform distribution, creating a grid that may be invaluable for some users.  It was superseded by later databases only due to their more comprehensive star listing, but they lose the spacial grid.  This one should be retained in the catalog system.

These catalogs should be available from the original CD.  Otherwise, they can be found in rpms for openSUSE on the website software.opensuse.org.  If they are added, then it would make sense to add them the same way the GSC2201 is set up.

**Comment by rmathar** [comment]
I failed to find the USNO SA2.0 on the opensuse server. The only indication of existance 
is a RPM packet on https://build.opensuse.org/package/show/home%3Amsalle%3Aextras/xephem .
If you process this with rpm2cpio (or download the xephem_usno.tar.xz directly
und unxz, untar the file) this results in 20 files named zone0300.cat ... zone1725.cat
(and a sort of index files named zone*.acc). The problems are
i) files  from zone0000.cat up to zone0225.cat are missing.
  They can probably be recovered from https://ftp.imcce.fr/pub/catalogs/USNO-SA2/
  or from http://catalogs.astro.uni-altai.ru/usnosa/ 
ii) Each *.cat-file should have a length that is a multiple of 3 x 4=12 bytes, if they
  are in the binary USNO A2.0 format, see https://ftp.imcce.fr/pub/catalogs/USNO-SA2/readuse.v10 .
  They actually have the  sizes as in https://www.projectpluto.com/a2.htm#sa2 .
  So apparently one does not  use the hints of https://www.projectpluto.com/a2.htm#compressing
  to retrieve the 3 x 4 bytes per star.

It is suspicious that xephem_usno.tar.xz has a size of
299,993,604 bytes (=286.1 MiB). It looks like some intermediate software has
chopped this at a strict 300,000,000 bytes limit.

Note that the USNO SA2.0 is no longer available on the USNO server itself.
(apparently only some interactive web-based retrieval tools for tiny
patches of the sky are supported?).

[Apparently only way to reconstruct the USNO SA2.0 is to fetch
the full USNO A2.0 from https://cdsarc.cds.unistra.fr/viz-bin/cat/I/252
and to write a program according to the recipes of  Dave Monet in the readmesa.v10
file of the RPM to reduce the number of stars to approximately 10 percent.
(The newbin.tar file with the original Fortran sources may not be available anywhere.)]

[The ftp and URL's of the Vizir catalog have long expired and there is
only limited hope that the original programs that were used are
still somewhere; likewise the link to the ftp directory in I/252
of the Vizir server is broken, you need to jump to https://cdsarc.u-strasbg.fr/ftp/I/252/
... So recompiling the src/ppm program for your computer would probably
allow to convert the *.ppm files of the VIZIR catalog to something readable.
Alternatively https://www.projectpluto.com/access.htm#A10_extractor]

In summary: there is no indication so far that the USNO SA2.0 has ever been on an
openSUSE server in the xephem format.

**Comment by rmathar** [comment]
The total size of the 24 USNO SA2.0 files in their binary format is 664418868 bytes with 55368239 objects.
If compiled into the ASCII XEphem format, the result is approximately 3 times that big, 1900 MiB.
I'll post a scanner/converter in https://vixra.org/abs/1802.0035 .

**Comment by dave-kaye** [comment]
I got complete XEphem downloads from multiple locations on openSUSE with several people packaging the product. One link is
http://download.opensuse.org/repositories/home:/lemmy04/openSUSE_Tumbleweed/src/xephem-4.1.0-1.6.src.rpm

It is a source rpm file, so either unpack it with newer versions of Ark or run the rpm command rpm -i filename.  The resulting directories will contain a tar/gzip file of several different catalogs.

I went through all the locations on openSUSE I could find, they all have the same set of files when the USNO section is expanded.  So the original file came from someplace.  I am not familiar with the available catalogs since I use the free version and never had access to the commercial version with the additional catalogs.  However, on this GitHub site in the Official directory exists a file ReadMe.txt.  It states that a USNO SA2.0 image catalog is in the commercial version of XEphem 3.7, along with other field star catlogs.  All of the listed field star catalogs show up in the openSUSE software, with the same naming structure as in the ReadMe.  I made an assumption that this is from the commercial product - that may be in error.

I have re-looked at the file and indeed do believe that there are zones 0, 75, 150 and 225 that are missing.  I had thought that there was some reason for this if it came from the commercial product.  As to the internal structure of the files, I have not looked at the files nor the XEphem file reading code.

If it is not a complete version, then the commercial cd could/should have it.  Maybe need to contact someone about getting a copy of the cd.  If it is only a partial on that cd, then need to look elsewhere.  Maybe a ClearSky contact could find the complete catalog for Xephem.

I would not put forth the effort to re-create the catalog from scratch or try to use more recent catalogs to generate a newer version.  The xephem_usno.tar.xz contained readmesa.v10 file states: "This CD-ROM contains USNO-SA1.0, the spatially sub-sampled version of USNO-A1.0 created to provide an astrometric reference catalog for asteroid hunters and other folks who do not need the entire USNO-A catalog."  So if it is available, then I had proposed keeping it in the catalog tree, since it does seem to fill a unique need.

**Comment by dave-kaye** [comment]
Sorry, I have been running in multiple directions the past several days. so I did not have time to look for the USNO-SA2.0 catalog.  The link from rmathar "They can probably be recovered from https://ftp.imcce.fr/pub/catalogs/USNO-SA2/" does appear to give a complete catalog.  The files do appear to be the same as in openSUSE except they are zone complete.  I will 1) attempt to get downloads and do a bit comparison on several of the files to see if they are the same  2) see if I can contact the openSUSE packagers for more information on where that file was generated.  If they are the same, then there is no need to modify the files to ASCII format. The files in openSUSE were being accessed by XEphem using the field star data system.

**Comment by rmathar** [comment]
Now I see. XEphem has dedicated scanners for some field star catalogs (i.e., the source code in ```GUI/xephem/usno.c```) such that just copying the '```*.cat``` and ```*.acc'``` files into the directory ```catalogs/usno/``` suffices. So I was wrong arguing that the USNO A2.0 binary files need to be converted to the generic ASCII format of the catalogs for use with XEphem.

**Comment by dave-kaye** [comment]
I have downloaded the files from https://ftp.imcce.fr/pub/catalogs/USNO-SA2/ and the md5 information file from http://catalogs.astro.uni-altai.ru/usnosa/.

Both the *.fr and openSUSE files matched the md5 values.  The bit comparison for both sources matched for all files.  The openSUSE files are the original USNO SA2.0 files, minus 8 missing files for 4 zones.

I loaded the catalog using:
Data --> Field Stars --> USNO A or SA catalogs (directory location)
Since these are zones and I do not know what zones are where, I randomly accessed sections of the sky and verified that field stars appear/disappear as expected with the XEphem controls.  It is assumed that all the zones will therefore work.

**Comment by dave-kaye** [comment]
Started fork to add the USNO-SA2.0 catalog, but hit a major roadblock.  Can add files only up to 25 MB via web access, and some files are larger.  Would someone add the *.cat files for zones 0375, 0450, 0525, 0600, 0675, 0750, 0825, 0900, 0975, 1050, 1125, 1200, 1275 and 1350.  They can be downloaded from the *.fr links above.

They would have to be downloaded and added via the command line git or local git.  The networked computers run Windows and I don't have admin access to allow the git software.

**Comment by dave-kaye** [comment]
I contacted Downey and he is not active in XEphem and could not remember the source of the USNO-SA2.0 files.

I contacted one of the openSUSE packagers.  The file should have come from an earlier release of XEphem 3.7 (3.6?) series, and the files were missing probably due to CD size limitations.

Both the readme file in the catalog directory and the help file show this catalog as part of the distribution.

**Comment by rmathar** [comment]
In https://github.com/dave-kaye/Catalogs (i) I moved usno to Official/usno to comply with the directory layout of the main branch, and (ii) added the full set of ```*cat``` files as taken from the french site. (Succession of git clone, a loop over wget's to https://ftp.imcce.fr/pub/catalogs/USNO-SA2/ , local moves, git add, git commit and git push....). I'll leave it to someone else to create a push request...

**Comment by dave-kaye** [comment]
Thank you for the updates.

I will download the new directories and verify the checksums/bit compare to be sure there was not a transfer error.  Then I will create a pull request, probably tomorrow.
